components/event-gateway/pkg/nats/server.go

```diff
@@ -167,8 +172,11 @@ func Spawn(c *config.EventGatewayConfig) error {
 		"port": c.Service.Port,
 	}).Info("Starting NATS messaging Server for event-gateway")
 
-	ns.Start()
-	return errors.New("NATS server exited")
+	go ns.Start()
```

Comment:
since we are adding the statements to stop the control, do we still need to define it as a routine?
can you test by running without routine.